packages/MarkdownEditor-Core.package/MarkdownReadStream.class/instance/next..st

```diff
@@ -0,0 +1,11 @@
+accessing
+next: anInteger
+
+	| ans endPosition |
+
+	endPosition := position + anInteger  min:  readLimit.
+	ans := collection copyFrom: position+1 to: endPosition.
+	(ans size ~= anInteger) 
+		ifTrue: [ans := ans padded: #right to: anInteger with: Character space].
```

Comment:
Line ends and beginnings are supposed to be seen as spaces. Thats why we do it.

Reply:
Yeah, but we still get 5 spaces, if I peek for the next 8 characters and there are only 3 left in the stream, right? I'm not sure whether we should have that, but I guess it does not matter for now if we do. It just might be a point of confusion for other people, if we do not necessarily need it but they might think we do

Reply:
I'd like to see a quote from the specs where it does say that eof should be treated like spaces. Afaik eol or beginning of line should be treated like whitespace which is specific to emphasis and can be implemented in the stream, but I still have vague feelings for it

Reply:
You're right, both EOL and Beginning Of Line shall be treated as Unicode Whitespace:
[GFM Spec Flanking](https://github.github.com/gfm/#left-flanking-delimiter-run)
> For purposes of this definition, the beginning and the end of the line count as Unicode whitespace.

Because as of now only the `MarkdownInlineParser` uses the `MarkdownReadStream` and because it shall only be passed text line by line (or multiple lines being part of exactly one paragraph), we decided to put the method into the stream.

Reply:
We changed it so that one space is added but if `next:` is called with an integer that would require adding more than one space an Error is raised.